[RFC] vim-patch:7.4.{1952, 1990, 2033, 2284}

KillTheMule opened the pull request:
Closes https://github.com/neovim/neovim/issues/5247. Also added the documentation change from https://github.com/vim/vim/commit/802a0d902fca423acb15f835d7b09183883d79a0.
## vim-patch:7.4.1952

> Problem:    Cscope interface does not support finding assignments.
> Solution:   Add the "a" command. (ppettina, closes vim/vim#882)

vim/vim@b12e7ef

All changes applied manually.
## vim-patch:7.4.1990

> Problem:    Cscope items are not sorted.
> Solution:   Put the new "a" command first. (Ken Takata)

vim/vim@80632db

All changes applied manually.
## vim-patch:7.4.2033

> Problem:    'cscopequickfix' option does not accept new value "a".
> Solution:   Adjust list of command characters. (Ken Takata)

vim/vim@6d20e17

All changes applied manually. Definition of `CSQF_CMDS` was moved to
`option_defs.h` in nvim, so changes applies there instead of `options.h`.
## vim-patch:7.4.2284

> Problem:    Comment in scope header file is outdated. (KillTheMule)
> Solution:   Point to the help instead. (closes #1017)

https://github.com/vim/vim/commit/f4145d8e990a72bdfea9db3110a7e42a0ff4240c

KillTheMule committed:
vim-patch:7.4.1952

Problem:    Cscope interface does not support finding assignments.
Solution:   Add the "a" command. (ppettina, closes vim/vim#882)

https://github.com/vim/vim/commit/b12e7ef956e0b0344778b7ef93d41f4b4ed2a670

All changes applied manually.

KillTheMule committed:
vim-patch:7.4.1990

Problem:    Cscope items are not sorted.
Solution:   Put the new "a" command first. (Ken Takata)

https://github.com/vim/vim/commit/80632db65e8f5f775dadbbc10c5ba6c173ebb24f

All changes applied manually.

KillTheMule committed:
vim-patch:7.4.2033

Problem:    'cscopequickfix' option does not accept new value "a".
Solution:   Adjust list of command characters. (Ken Takata)

https://github.com/vim/vim/commit/6d20e1754461b0f8d395f2e3464f0dc1060497f7

All changes applied manually. Definition of `CSQF_CMDS` was moved to
`option_defs.h` in nvim.

KillTheMule committed:
Lint

justinmk commented on src/nvim/if_cscope.c:
```diff
@@ -970,6 +970,9 @@ static int cs_find_common(char *opt, char *pat, int forceit, int verbose,
     break;
+  case '9':
+    cmdletter = 'a';
+    break;
```
in the original patch this case was also added to `cs_create_cmd`

KillTheMule replied:
Wooops, sorry for that, fixed!

oni-link commented:
In vim/vim@802a0d902fca423acb15f835d7b09183883d79a0 the documentation for `if_cscop.txt` is updated.

KillTheMule commented:
That's a runtime update, and we're very far from merging it. I propose I merge the relevant doc changes now, the one to port the runtime patch will notice and easily work around.

KillTheMule committed:
Add if_cscope documentation for querytype 'a'

From 802a0d902fca423acb15f835d7b09183883d79a0.

oni-link commented:
A comment block could also be updated [here](https://github.com/neovim/neovim/blob/master/src/nvim/if_cscope_defs.h#L38-Lundefined).

KillTheMule commented:
Good catch, I updated an opend a [vim issue](https://github.com/vim/vim/issues/1017) so they can fix it as well.

justinmk commented:
I guess https://github.com/vim/vim/commit/f4145d8e990a72bdfea9db3110a7e42a0ff4240c should be included instead of the comment update.

KillTheMule committed:
vim-patch:7.4.2284

Problem:    Comment in scope header file is outdated. (KillTheMule)
Solution:   Point to the help instead. (closes vim/vim#1017)

https://github.com/vim/vim/commit/f4145d8e990a72bdfea9db3110a7e42a0ff4240c

KillTheMule commented:
Thanks, justinmk, I included that patch.

KillTheMule commented:
CI Failures are from master (why doesn't QB fail? It _does_ run the test).

oni-link commented on src/nvim/version.c:
```diff
@@ -75,6 +75,7 @@ static char *features[] = {
 // clang-format off
 static int included_patches[] = {
+  2284,
   2219,
```
We are missing some patch number comments from 2200 to 2284.

fwalch commented:
> CI Failures are from master (why doesn't QB fail? It does run the test).

Travis seems to run https://github.com/neovim/neovim/commit/b6a360fb4037275f3eb572c25f366ef06afef95d, whereas QB ran https://github.com/neovim/neovim/commit/89a7a4f02c35e414bc0bd33560738c9cf0abe33e
